File: 05/Java/TripleExpression/Add.java
package TripleExpression;

public class Add extends BinaryOperator implements TripleExpression {
    public Add(TripleExpression first, TripleExpression second) {
        super(first, second);
    }

    protected double apply(double a, double b) {
        return a + b;
    }
}

File: 05/CSharp/TripleExpression_Classes_ONLY/Add.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Add : BinaryOperator
    {
        public Add(ITripleExpression first, ITripleExpression second)
        {
            fOp = first;
            sOp = second;
        }
        
        protected override double Apply(double a, double b)
        {
            return a + b;
        }
    }
}
